src/xds/client.rs

```diff
@@ -274,16 +276,28 @@ impl AdsClient {
             .into_inner();
 
         info!("Stream established");
+        let (tx, rx) = oneshot::channel();
+        let mut initial_xds_tx = Some(tx);
+        let initial_xds_rx = Some(rx);
+        let ready = mem::take(&mut self.block_ready);
+        tokio::spawn(async move {
+            match initial_xds_rx.unwrap().await {
+                Ok(_) => {
+                    drop(ready);
```

Comment:
cannot drop self.block_ready?

Reply:
Unfortunately not; `run_internal` has a mutable reference for `self` so I can't drop a field on self

Reply:
Maybe pass `block_ready` into the `run`

Reply:
I thought about that, but the `block_ready` would have to be passed into `WorkloadManager::run` (which is a bit awkward from the caller's perspective). Furthermore, if we want to avoid `/admin/ready` flip-flopping (i.e. ready in one request and then not ready in a subsequent request), it's better to call `register_task`s around the same time. Waiting until `WorkloadManager::run` to create a `block_ready` for ADS when the LocalClient's already exists opens the door for that kind of inconsistent behavior.